Iconography.js

```diff
@@ -0,0 +1,166 @@
+import * as DOM from "AXM/DOM";
+import * as AXMUtils from "AXM/AXMUtils";
+
+class Decorator {
+    constructor(
+        name,
+        [y, x] = orientation,
+        [offsetX, offsetY] = offset,
+        { size, opacity, color } = options
+    ) {
+        this.name = name;
+
+        if (["left", "right"].indexOf(x) < 0) {
+            AXMUtils.Test.reportBug("Decorator x position should be 'left' or 'right'");
+        }
+
+        if (["top", "bottom"].indexOf(y) < 0) {
+            AXMUtils.Test.reportBug("Decorator y position should be 'top' or 'bottom'");
+        }
+
+        this.xPos = x;
+        this.yPos = y;
+        this.offsetX = offsetX;
+        this.offsetY = offsetY;
+
+        this.size = size;
+        this.opacity = opacity;
+        this.color = color;
+    }
+}
+
+class IconStack {
+    constructor(icons = [], { scaling } = { scaling: "1x" }) {
+        if (icons.length < 0) {
+            throw Error(`must have at least one icon to stack`);
+        }
+
+        this.icons = icons;
+
+        const validFormFactors = ["lg", "1x", "2x", "3x", "4x", "5x"];
+        if (!validFormFactors.includes(scaling)) {
+            throw Error(`scaling factor must be one of ${validFormFactors.join(", ")}`);
+        }
+
+        this.scaling = scaling;
+    }
+
+    renderHtml() {
+        return DOM.Span(
+            { className: `fa-stack fa-lg` },
+            this.icons.map(icon => {
+                // icon.addClass(`fa-stack-${this.scaling}`);
+                return icon;
+            })
+        );
+    }
+}
+
+class Icon {
+    constructor({ name, decorators = [], style = {}, classNames = [ ] } = {}) {
+        Object.assign(this, AXMUtils.object("icon"));
+
+        this.name = name;
+        this._decorators = decorators;
+
+        this._sizeFactor = 1;
+
+        this.style = style;
+        this.classNames = [
+            `fa ${this.name}`,
+            ...classNames
+        ]
+    }
+
+    addClass(additionalClassName) {
+        this.classNames.push(additionalClassName);
+    }
+
+    overrideStyle(additionalStyling = { k, v }) {
```

Comment:
It would inform language server in IDEs about the form of the argument (i.e. an object with key value mapping), however in this case it's wrong to start with, so I'll go ahead and remove it.